### ISSUE TITLE
FR: Arrow-key functionality in SoarCLI

### ISSUE THREAD

**Issue by ShadowJonathan** [SoarGroup/Soar#447]
Currently exploring the soar CLI, and one ergonomic feature I'm missing from other shells is the ability to scroll through shell history with the up arrow, so that mistakes in commands can be easily altered with one up-press, which'd load the last-entered command in the current prompt.

Edit: Amended with more functionality, such as left-right and ctrl-A & ctrl-E (beginning and end)

**Comment by scijones** [comment]
I, too, have wanted this (and other) functionality I have come to expect in other shells.

**Comment by ShadowJonathan** [comment]
What other functionality, for the record? (Just so it's noted somewhere, and can be spun out to another FR)

**Comment by garfieldnate** [comment]
Yeah, I wonder if there's a library somewhere we can use for this. I also want ctrl-A and ctrl-E for navigation to the beginning and end of lines (and I know there are more that are fairly standard). Actually, if I remember correctly, you can't even use the arrow keys to go back and forth in a line and edit it! Truly aggravating and also very surprising.

**Comment by scijones** [comment]
The main other one for me is what was just mentioned -- using the arrow keys to go back and forth in a line and edit it.

**Comment by scijones** [comment]
More info: When I run soar.exe within windows powershell, "up arrow to previous command" and "use left/right to change cursor position and then edit command" both work.

**Comment by garfieldnate** [comment]
That is interesting. I have no idea why that's the case!
